src/components/Questionnaire/QuestionnaireEditor.tsx

```diff
@@ -1112,36 +1147,329 @@ export default function QuestionnaireEditor({ id }: QuestionnaireEditorProps) {
+                id={id}
+                organizations={organizations}
+                organizationSelection={{
+                  selectedIds: selectedOrgIds,
+                  onToggle: handleToggleOrganization,
+                  searchQuery: orgSearchQuery,
+                  setSearchQuery: setOrgSearchQuery,
+                  available: availableOrganizations,
+                  isLoading: isLoadingAvailableOrganizations,
+                }}
+                tags={questionnaire.tags}
+                tagSelection={{
+                  selectedIds: selectedTagIds,
+                  onToggle: handleToggleTag,
+                  searchQuery: tagSearchQuery,
+                  setSearchQuery: setTagSearchQuery,
+                  available: availableTags,
+                  isLoading: isLoadingAvailableTags,
+                }}
+                activeTab={activeTab}
               />
-            </CardContent>
-          </Card>
+            </div>
+          </div>
+          <DebugPreview
+            data={questionnaire}
+            title={t("questionnaire")}
+            className="mt-4"
+          />
         </TabsContent>
       </Tabs>
     </div>
   );
 }
+function QuestionPreview(question: Question, index: number, depth: number) {
+  const isRequired = question.required;
+  const hasSubQuestions =
+    question.type === "group" && (question.questions?.length ?? 0) > 0;
+
+  return (
+    <div className="space-y-2">
+      <div
+        className={`pl-3 -ml-6 ${
+          depth === 0
+            ? isRequired
+              ? "border-l-4 border-purple-500"
+              : "border-l-4 border-gray-300"
+            : ""
+        }`}
+      >
+        <div className="flex items-center justify-between">
+          <span className="text-gray-700 font-bold">
+            {question.text || t("Untitled")}
+            {isRequired && <span className="text-red-500 ml-1">*</span>}
+          </span>
+          <Button variant="ghost" className="underline">
+            + {t("add_notes")}
+          </Button>
+        </div>
+      </div>
+
+      {question.type === "boolean" && (
+        <RadioGroup>
+          <div className="flex gap-4">
+            <div className="flex items-center space-x-2">
+              <RadioGroupItem value="true" id={`boolean-yes-${question.id}`} />
+              <Label htmlFor={`boolean-yes-${question.id}`}>{t("yes")}</Label>
+            </div>
+            <div className="flex items-center space-x-2">
+              <RadioGroupItem value="false" id={`boolean-no-${question.id}`} />
+              <Label htmlFor={`boolean-no-${question.id}`}>{t("no")}</Label>
+            </div>
+          </div>
+        </RadioGroup>
+      )}
+
+      {(question.type === "decimal" || question.type === "integer") && (
+        <Input
+          type="number"
+          className="border border-gray-300 rounded-md p-2 text-sm w-full"
+        />
+      )}
+
+      {question.type === "structured" && (
+        <Input
```

Comment:
Are you sure this is accurate? 🤔  Each of the structured ones are custom questions, shouldn't be just a simple input.